code/modules/mob/living/carbon/human/species/moth.dm

```diff
@@ -0,0 +1,229 @@
+		restorewings(H)
+
+/datum/species/moth/spec_movement_delay(mob/living/carbon/H)
+	return FALSE
+
+/**
+ * Copies wing and antennae names to species datum vars
+ */
+/datum/species/moth/proc/backupwings(mob/living/carbon/human/H)
+	var/obj/item/organ/external/head/A = H.get_organ("head")
+	pre_burn_antennae = A.ha_style
+	pre_burn_wings = H.body_accessory.name
+
+/**
+ * Sets wings and antennae to burnt variants, removing some species buffs
+ */
+/datum/species/moth/proc/destroywings(mob/living/carbon/human/H)
+	burnt_wings = TRUE
+	backupwings(H)
+	H.change_body_accessory("Burnt Off Wings")
+	H.change_head_accessory("Burnt Off Antennae")
+
+/**
+ * Restores wings and antennae from values in species datum vars
+ */
+/datum/species/moth/proc/restorewings(mob/living/carbon/human/H)
+	burnt_wings = FALSE
+	H.change_head_accessory(pre_burn_antennae)
+	H.change_body_accessory(pre_burn_wings)
+	pre_burn_antennae = null
+	pre_burn_wings = null
+
+/datum/action/innate/cocoon
+	name = "Cocoon"
+	desc = "Restore your wings and antennae, and heal some damage. If your cocoon is broken externally you will take heavy damage!"
+	check_flags = AB_CHECK_CONSCIOUS
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "cocoon1"
+
+/datum/action/innate/cocoon/Activate()
+	var/mob/living/carbon/human/moth/H = owner
+	H.visible_message("<span class='notice'>[H] begins to hold still and concentrate on weaving a cocoon...</span>", "<span class='notice'>You begin to focus on weaving a cocoon... (This will take [COCOON_WEAVE_DELAY] seconds, and you must hold still.)</span>")
+	if(do_after(H, COCOON_WEAVE_DELAY, FALSE, H))
+		if(H.incapacitated(ignore_lying = TRUE))
+			to_chat(H, "<span class='warning'>You cannot weave a cocoon in your current state.</span>")
+			return
+		H.visible_message("<span class='notice'>[H] finishes weaving a cocoon!</span>", "<span class='notice'>You finish weaving your cocoon.</span>")
+		var/obj/structure/moth/cocoon/C = new(get_turf(H))
+		H.forceMove(C)
+		C.preparing_to_emerge = TRUE
+		var/datum/species/moth/M = H.dna.species
+		M.cocooned = TRUE
+		H.KnockOut()
+		addtimer(CALLBACK(src, .proc/emerge, C), COCOON_EMERGE_DELAY, TIMER_UNIQUE)
+	else
+		to_chat(H, "<span class='warning'>You need to hold still in order to weave a cocoon!</span>")
+
+/**
+ * Removes moth from cocoon, heals moth and restores burnt wings
+ */
+/datum/action/innate/cocoon/proc/emerge(obj/structure/moth/cocoon/C)
+	for(var/mob/living/carbon/human/H in C.contents)
+		var/datum/species/moth/M = H.dna.species
+		M.cocooned = FALSE
+		H.adjustBruteLoss(COCOON_HEAL_AMOUNT)
+		H.adjustFireLoss(COCOON_HEAL_AMOUNT)
+		if(M.burnt_wings)
+			M.restorewings(H)
+	C.preparing_to_emerge = FALSE
+	qdel(C)
+
+/obj/structure/moth/cocoon
+	name = "moth cocoon"
+	desc = "Someone wrapped in a moth cocoon"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "cocoon1"
+	color = COLOR_PALE_YELLOW //So tiders (hopefully) don't decide to immediately bust them open
```

Comment:
not quite sure about this comment? how does yellow make tiders not attack?

Reply:
If it's white, they will assume it's a spider cocoon, because they probably don't bother to read the status bar. Yellow hopefully differentiates it from regular cocoons enough